ceremonies/meetup-validation/src/tests.rs

```diff
@@ -0,0 +1,228 @@
+		vec![vec![1, 2, 4], vec![1], vec![0, 1, 4], vec![0, 1, 2, 4], vec![0, 1, 2]];
+	let excluded_participants: Vec<(usize, ExclusionReason)> = vec![
+		(3, ExclusionReason::TooFewIncomingAttestations),
+		(1, ExclusionReason::TooFewOutgoingAttestations),
+	];
+	assert_eq!(
+		get_excluded_participants_num_attestations(
+			&participants,
+			&participant_attestations,
+			|n| n - 1
+		),
+		excluded_participants
+	);
+}
+
+#[test]
+fn find_majority_vote_works() {
+	let participants: Vec<usize> = vec![0, 1, 2, 3, 4];
+	let participant_votes: Vec<u32> = vec![1, 1, 2, 3, 1];
+	assert_eq!(find_majority_vote(&participants, &participant_votes), Ok((1u32, 3u32)));
+}
+
+#[test]
+fn get_updated_participants_works_case_0() {
+	// Scenario: Everyone attests everyone
+	let participants: Vec<usize> = vec![0, 1, 2];
+	let participant_votes: Vec<u32> = vec![3, 3, 3];
+	let participant_attestations: Vec<Vec<usize>> = vec![vec![1, 2], vec![0, 2], vec![0, 1]];
+
+	let updated_participants =
+		UpdatedParticipants { included: participants.clone(), excluded: vec![] };
+	let attestation_threshold_fn = |i| i - 1;
+
+	assert_eq!(
+		get_updated_participants(
+			&participants,
+			&participant_votes,
+			&participant_attestations,
+			attestation_threshold_fn,
+		)
+		.unwrap(),
+		updated_participants
+	);
+}
+
+#[test]
+fn get_updated_participants_works_case_1() {
+	// Scenario: One participant did not vote
+	let participants: Vec<usize> = vec![0, 1, 2, 3];
+	let participant_votes: Vec<u32> = vec![0, 3, 3, 3];
+	let participant_attestations: Vec<Vec<usize>> =
+		vec![vec![1, 2, 3], vec![0, 2, 3], vec![0, 1, 3], vec![0, 1, 2]];
+
+	let updated_participants = UpdatedParticipants {
+		included: vec![1, 2, 3],
+		excluded: vec![ExcludedParticipant { index: 0, reason: ExclusionReason::NoVote }],
+	};
+	let attestation_threshold_fn = |i| i - 1;
+
+	assert_eq!(
+		get_updated_participants(
+			&participants,
+			&participant_votes,
+			&participant_attestations,
+			attestation_threshold_fn,
+		)
+		.unwrap(),
+		updated_participants
+	);
+}
+
+#[test]
+fn get_updated_participants_works_case_2() {
+	// Scenario: One participant did not vote like majority
+	let participants: Vec<usize> = vec![0, 1, 2, 3];
+	let participant_votes: Vec<u32> = vec![1, 3, 3, 3];
+	let participant_attestations: Vec<Vec<usize>> =
+		vec![vec![1, 2, 3], vec![0, 2, 3], vec![0, 1, 3], vec![0, 1, 2]];
+
+	let updated_participants = UpdatedParticipants {
+		included: vec![1, 2, 3],
+		excluded: vec![ExcludedParticipant { index: 0, reason: ExclusionReason::WrongVote }],
+	};
+	let attestation_threshold_fn = |i| i - 1;
+
+	assert_eq!(
+		get_updated_participants(
+			&participants,
+			&participant_votes,
+			&participant_attestations,
+			attestation_threshold_fn,
+		)
+		.unwrap(),
+		updated_participants
+	);
+}
+
+#[test]
+fn get_updated_participants_works_case_3() {
+	// Scenario: 5 participants
+	// 0 has a broken phone and can only be attested by 1 and 2 and attests nobody
```

Comment:
what exactly is wrong with this phone? camera not working? I think you're combining two problems here: `0` doesn't attest anyone and is not attested by some. I'd separate these cases